src/api/MainSolver.cc

```diff
@@ -559,12 +560,32 @@ void MainSolver::Preprocessor::initialize() {
     substitutions.push();
 }
 
+void MainSolver::Preprocessor::prepareForProcessingFrame(std::size_t frameIndex) {
+    assert(frameIndex < virtualFrameCount);
+    while (actualFrameCount <= frameIndex) {
+        pushInternal();
+    }
```

Comment:
`assert(actualFrameCount == virtualFrameCount);` ?

Reply:
That does not hold, `virtualFrameCount` can still be larger.

Reply:
Right ...